code/game/objects/items/gift.dm

```diff
@@ -0,0 +1,251 @@
+/* Gifts and wrapping paper
+ * Contains:
+ * Gifts
+ * Wrapping Paper
+ */
+
+/*
+ * Gifts
+ */
+
+GLOBAL_LIST_EMPTY(possible_gifts)
+
+/obj/item/a_gift
+	name = "gift"
+	desc = "A wrapped bundle of joy, you'll have to get closer to see who it's addressed to."
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "gift0"
+	resistance_flags = RESIST_ALL
+	///if true the present can be opened by anybody
+	var/freepresent = FALSE
+	///who is the present addressed to?
+	var/mob/living/carbon/human/present_receiver = null
+	///item contained in this gift
+	var/obj/item/contains_type
+	///real name of the present receiver
+	var/present_receiver_name = null
+
+/obj/item/a_gift/Initialize(mapload)
+	. = ..()
+	pixel_x = rand(-10,10)
+	pixel_y = rand(-10,10)
+	icon_state = "gift[rand(0,10)]"
+
+	contains_type = get_gift_type()
+
+/obj/item/a_gift/examine(mob/user)
+	. = ..()
+	if(present_receiver)
+		. += "This present is addressed to [present_receiver_name]."
+
+/obj/item/a_gift/attack_self(mob/M)
+	if(present_receiver == null && !freepresent)
+		get_recipient() //generate owner of gift
+	to_chat(M, span_warning("You start unwrapping the present, trying to locate any sign of who the present belongs to..."))
+	if(!do_after(M, 4 SECONDS))
+		return
+	if(!freepresent && present_receiver != M)
+		if(tgui_alert(M, "This present is addressed to [present_receiver_name]. Open it anyways?", "Continue?", list("Yes", "No")) != "Yes")
+			M.visible_message(span_notice("[M] tears into [present_receiver_name]'s gift with reckless abandon!"))
+			M.balloon_alert_to_viewers("Open's [present_receiver_name]'s gift" ,ignored_mobs = M)
+			log_game("[M] has opened a present that belonged to [present_receiver_name] at [AREACOORD(loc)]")
+			if(prob(70) || HAS_TRAIT(M, TRAIT_CHRISTMAS_GRINCH))
+				GLOB.round_statistics.presents_grinched += 1
+				if(!HAS_TRAIT(M, TRAIT_CHRISTMAS_GRINCH))
+					GLOB.round_statistics.number_of_grinches += 1
+					ADD_TRAIT(M, TRAIT_CHRISTMAS_GRINCH, TRAIT_CHRISTMAS_GRINCH) //bad present openers are effectively cursed to receive nothing but coal for the rest of the round
+					to_chat(M, span_boldannounce("Your heart feels three sizes smaller..."))
+				spawnpresent(M) //they have the grinch trait, the presents will always spawn coal
+			else
+				spawnpresent(M, TRUE) //they got lucky, the present will open as normal but with a STOLEN label in the desc
+			qdel(src)
+		return
```

Comment:
inverting this check would do a lot to making it a lot more readable, if you care about (considering this is a xmas PR)